ecl/wutest/wutest.cpp

```diff
@@ -755,6 +745,123 @@ class WuTest : public CppUnit::TestFixture
         factory->deleteWorkUnit(wuid);
     }
 
+    void testGraph()
+    {
+        Owned<IWorkUnitFactory> factory = getWorkUnitFactory();
+        Owned<IWorkUnit> createWu = factory->createWorkUnit("WuTest", NULL, NULL, NULL);
+        StringBuffer wuid(createWu->queryWuid());
+        createWu->createGraph("Graph1", "graphLabel", GraphTypeActivities, createPTreeFromXMLString("<graph/>"));
+        createWu->createGraph("Graph2", "graphLabel", GraphTypeActivities, createPTreeFromXMLString("<graph/>"));
+        createWu->createGraph("Graph3", "graphLabel", GraphTypeEcl, createPTreeFromXMLString("<graph/>"));
+        createWu->setState(WUStateCompleted);
+        createWu->commit();
+        createWu.clear();
+        // Now try to reread a single graph....
+        SCMStringBuffer s;
+        Owned<IConstWorkUnit> wu = factory->openWorkUnit(wuid);
+        ASSERT(streq(wu->queryWuid(), wuid));
+        Owned<IConstWUGraph> graph = wu->getGraph("Graph1");
+        ASSERT(graph != NULL);
+        ASSERT(graph->getType() == GraphTypeActivities);
+        ASSERT(streq(graph->getName(s).str(),"Graph1"));
+        ASSERT(streq(graph->getLabel(s).str(),"graphLabel"));
+        ASSERT(streq(graph->getXGMML(s, false).str(), "<graph/>\n"));
+
+        // Then the lightweight meta....
+        wu.setown(factory->openWorkUnit(wuid));
+        ASSERT(streq(wu->queryWuid(), wuid));
+        Owned<IConstWUGraphMetaIterator> it = &wu->getGraphsMeta(GraphTypeActivities);
+        unsigned numIterated = 0;
+        ForEach (*it)
+        {
+            ASSERT(it->query().getType() == GraphTypeActivities);
+            ASSERT(streq(graph->getLabel(s).str(),"graphLabel"));
+            numIterated++;
+        }
+        ASSERT(numIterated==2);
+
+        wu.setown(factory->openWorkUnit(wuid));
+        ASSERT(streq(wu->queryWuid(), wuid));
+        it.setown(&wu->getGraphsMeta(GraphTypeAny));
+        numIterated = 0;
+        ForEach (*it)
+        {
+            ASSERT(streq(it->query().getLabel(s).str(),"graphLabel"));
+            numIterated++;
+        }
+        ASSERT(numIterated==3);
+
+        // then the heavy meta
+        wu.setown(factory->openWorkUnit(wuid));
+        ASSERT(streq(wu->queryWuid(), wuid));
+        Owned<IConstWUGraphIterator> it2 = &wu->getGraphs(GraphTypeActivities);
+        numIterated = 0;
+        ForEach (*it2)
+        {
+            ASSERT(it2->query().getType() == GraphTypeActivities);
+            ASSERT(streq(it2->query().getLabel(s).str(),"graphLabel"));
```

Comment:
It is, though you'd have to know the internal implementations to know it. I did check the tracing to be sure it fetched the full information in this case.